src/neuroconv/datainterfaces/ophys/inscopix/inscopiximagingdatainterface.py

```diff
@@ -0,0 +1,194 @@
+
+        # Subject information
+        subject_metadata = {}
+        has_any_subject_data = False
+
+        # Subject ID
+        if subject_info and subject_info.get("animal_id"):
+            subject_metadata["subject_id"] = subject_info["animal_id"]
+            has_any_subject_data = True
+
+        species_value = None
+        strain_value = None
+
+        if subject_info and subject_info.get("species"):
+            species_raw = subject_info["species"]
+            # If it contains genotype info or matches NWB format, put it in species; otherwise strain
+            if " " in species_raw and species_raw[0].isupper() and species_raw.split()[1][0].islower():
+                species_value = species_raw
+            else:
+                strain_value = species_raw
+
+            if species_value:
+                subject_metadata["species"] = species_value
+                has_any_subject_data = True
+            if strain_value:
+                subject_metadata["strain"] = strain_value
+                has_any_subject_data = True
+
+        # Sex mapping
+        sex_mapping = {"m": "M", "male": "M", "f": "F", "female": "F", "u": "U", "unknown": "U"}
+        if subject_info and subject_info.get("sex"):
+            mapped_sex = sex_mapping.get(subject_info["sex"].lower(), "U")
+            subject_metadata["sex"] = mapped_sex
+            has_any_subject_data = True
+
+        # Additional subject fields
+        if subject_info:
+            if subject_info.get("description"):
+                subject_metadata["description"] = subject_info["description"]
+                has_any_subject_data = True
+            if subject_info.get("date_of_birth"):
+                subject_metadata["date_of_birth"] = subject_info["date_of_birth"]
+                has_any_subject_data = True
+            if subject_info.get("weight") and subject_info["weight"] > 0:
+                subject_metadata["weight"] = str(subject_info["weight"])
+                has_any_subject_data = True
+
+        # Add Subject if we have ANY subject information, filling required fields with defaults
+        if has_any_subject_data:
+            if "subject_id" not in subject_metadata:
+                subject_metadata["subject_id"] = "Unknown"
+            if "species" not in subject_metadata:
+                subject_metadata["species"] = "Unknown species"
+            if "sex" not in subject_metadata:
+                subject_metadata["sex"] = "U"
+
+            if "Subject" in metadata and isinstance(metadata["Subject"], dict):
+                metadata["Subject"].update(subject_metadata)
+            else:
+                metadata["Subject"] = subject_metadata
+
+        return metadata
+
+    def add_to_nwbfile(
+        self,
+        nwbfile,
+        metadata: Optional[dict] = None,
+        photon_series_type: Literal["OnePhotonSeries", "TwoPhotonSeries"] = "OnePhotonSeries",
```

Comment:
You should add kwargs here and proapgated them to the super().add_to_nwbfile(, **kwargs)
```suggestion
        **kwargs,
```

Reply:
This way, the other kwargs from the add_to_nwbfile can be propagated.